### PR TITLE
Added loss graph to visualize training

### PR DESCRIPTION
**Pull request by r59q**
It is behind a feature flag. Create a cookie `fflags` and insert `loss-graph` as the value to enable (comma-separated for multiple values)

image

- Adds an onIteration function that can be handled. It will be supplied with the current epoch and what loss value it has reached
- Adds a chart.js line graph to visualize how loss changes as the model trains in real-time

**Thoughts**:
- Should we add some explanary texts to the page?
- Does the page become too cluttered with text? Maybe a restructuring is needed, especially with the KNN visualization
- Hasn't been tested with the KNN visualization, will probably run into some conflicts
- Adding buttons to change logarithmic/linear scales
- Since we can tap into the model at each iteration, we can compute accuracy in each iteration for a graph that may be easier to understand (this does take a lot of compute time though)
- Letting the users toy around with noOfEpochs, batchSize and learning rate could be introduced here
    -  If we do this, we could make the graph historic, saving the previous lines for comparison?

### CODE REVIEW

**Comment by Karlo-Emilo** [comment]
It looks good.

Improvements that could make it easier for students to read and decode the graph:
- Lock the x-axis if it is possible to reason about limits.
- Slow down the visualization of the graph. So make the training process artificially slower to give students more time to digest what is happening.

I do not think we should make these changes right now. We can consider them after we have discussed the graph with the Math teachers.

**Review by Karlo-Emilo — APPROVED** [review]
